deleter.py

```diff
@@ -344,6 +344,35 @@ def delete_parts(self, ids_to_remove):
             url = D42_URL + f
             r = requests.delete(url, headers=self.headers, verify=False)
             i += 1
+			
+    def delete_serviceinstances(self, ids_to_remove):
+        """
+		Added - 03/24/2020 - Not part of Wipe all
+        Deleting service instances (si) one at the time. Very slow!
+        :return:
+        """
+        if len(ids_to_remove) > 0:
+            all_ids = ids_to_remove
+        else:
+            all_ids = []
+            offset = 0
+            print ('\n[!] Deleting service instances')
+            while 1:
+                url = '/api/2.0/service_instances/?offset=%s' % offset
+                ids, offset, limit, total_count = self.get(url, 'serviceinstance_id', 'Service Instances')
```

Comment:
I don't think the parameters being passed to self.get are correct.  The ID field for the service_instances API is service_detail_id and the name of the property that the service instances come back in is service_details.  If you log into D42 and put this URL in another browser tab, you will see how the data is coming back:
https://[D42 Host]/api/2.0/service_instances/

Reply:
I am not sure what you are saying.. I did as you suggested and received a 504

Reply:
I updated the code as suggested:
This line:
ids, offset, limit, total_count = self.get(url, 'serviceinstance_id', 'Service Instances')
would need to be changed to this:
ids, offset, limit, total_count = self.get(url, 'service_detail_id', 'service_details')